NullPointerException in TextBlockGoogleStyleFormatting with text blocks in annotations

I have read check documentation: https://checkstyle.org/checks/coding/textblockgooglestyleformatting.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Example.java

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="TextBlockGoogleStyleFormatting"/>
  </module>
</module>
$ cat Example.java
@SuppressWarnings(
  """
  Text block in annotation.
  """)
public class Example {
}
$ RUN_LOCALE="-Duser.language=en -Duser.country=US"

$ java $RUN_LOCALE -jar checkstyle-12.3.0-all.jar -c config.xml Example.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing Example.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:313)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:227)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:429)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:347)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:206)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.NullPointerException: Cannot invoke "com.puppycrawl.tools.checkstyle.api.DetailAST.getType()" because "parent" is null
        at com.puppycrawl.tools.checkstyle.checks.coding.TextBlockGoogleStyleFormattingCheck.openingQuotesAreAloneOnTheLine(TextBlockGoogleStyleFormattingCheck.java:136)
        at com.puppycrawl.tools.checkstyle.checks.coding.TextBlockGoogleStyleFormattingCheck.visitToken(TextBlockGoogleStyleFormattingCheck.java:87)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:390)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:463)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:328)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:208)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:101)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:341)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:300)
        ... 5 more
Checkstyle ends with 1 errors.

```
---

I expect checkstyle to not throw report any issues in the code and not fail with a NullPointerException.

The above example is a minimal example using `@SuppressWarnings` since that is an annotation that is available without libraries. The problem was encountered using another annotation where using a text block was more meaningful. 

---

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/TextBlockGoogleStyleFormattingCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/TextBlockGoogleStyleFormattingCheck.java
@@ -150,7 +150,8 @@ public class TextBlockGoogleStyleFormattingCheck extends AbstractCheck {
                     TokenTypes.VARIABLE_DEF,
                     TokenTypes.METHOD_DEF,
                     TokenTypes.CTOR_DEF,
-                    TokenTypes.ENUM_DEF)) {
+                    TokenTypes.ENUM_DEF,
+                    TokenTypes.CLASS_DEF)) {
                 break;
             }
         }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/TextBlockGoogleStyleFormattingCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/TextBlockGoogleStyleFormattingCheckTest.java
@@ -372,4 +372,15 @@ public class TextBlockGoogleStyleFormattingCheckTest extends AbstractModuleTestS
         verifyWithInlineConfigParser(
                 getPath("InputTextBlockGoogleStyleFormatting10.java"), expected);
     }
+
+    @Test
+    public void testDefaultTextBlockFormatAnnotations() throws Exception {
+        final String[] expected = {
+            "59:27: " + getCheckMessage(MSG_OPEN_QUOTES_ERROR),
+            "66:19: " + getCheckMessage(MSG_OPEN_QUOTES_ERROR),
+            "68:1: " + getCheckMessage(MSG_VERTICALLY_UNALIGNED),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputTextBlockGoogleStyleFormattingAnnotations.java"), expected);
+    }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/textblockgooglestyleformatting/InputTextBlockGoogleStyleFormattingAnnotations.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/textblockgooglestyleformatting/InputTextBlockGoogleStyleFormattingAnnotations.java
@@ -1,0 +1,69 @@
+/*
+TextBlockGoogleStyleFormatting
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.textblockgooglestyleformatting;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@SuppressWarnings(
+        """
+        Text block in annotation
+        """
+)
+public class InputTextBlockGoogleStyleFormattingAnnotations {
+    public static void textFun() {}
+
+    @Target(ElementType.TYPE)
+    public @interface GetAnnon {}
+
+    @Target(ElementType.TYPE)
+    public @interface SetAnnon {}
+
+    @Target(ElementType.TYPE)
+    public @interface NamedNativeQuery{
+        String name();
+        String query();
+    }
+
+    @NamedNativeQuery(
+            name = "InventoryHost",
+            query =
+                    """
+                    h.id as inventory_id, h.org_id, h.modified_on, h.account, h.display_name,
+                    """)
+    @GetAnnon
+    @SetAnnon
+    class Inner {
+        public @interface Command{
+            String name();
+            String description();
+        }
+    }
+
+    @Inner.Command(
+            name = "Inner2",
+            description =
+                """
+                Create a private key and print the associated address on the blockchain
+                """
+    )
+    class Inner2 {}
+
+    @Inner.Command(
+            name = "Inner3",
+            // violation below 'Opening quotes (""") of text-block must be on the new line'
+            description = """
+                Create a private key and print the associated address on the blockchain
+                          """)
+    class Inner3 {}
+}
+
+// violation below 'Opening quotes (""") of text-block must be on the new line'
+@SuppressWarnings("""
+Text block in Annotation
+""") // violation 'Text-block quotes are not vertically aligned'
+class OuterFirst{}

